Test.System.Containers.Filesystem/CreateNewImageTests.cs

```diff
@@ -0,0 +1,118 @@
+using System.Containers.Tasks;
+using System.Diagnostics;
+
+namespace Test.System.Containers.Tasks;
+
+[TestClass]
+public class CreateNewImageTests
+{
+    [TestMethod]
+    public void CreateNewImage_Baseline()
+    {
+        DirectoryInfo newProjectDir = new DirectoryInfo(Path.Combine(Path.GetTempPath(), nameof(CreateNewImage_Baseline)));
+
+        if (newProjectDir.Exists)
+        {
+            newProjectDir.Delete(recursive: true);
+        }
+
+        newProjectDir.Create();
+
+        ProcessStartInfo info = new ProcessStartInfo
+        {
+            WorkingDirectory = newProjectDir.FullName,
+            FileName = "dotnet",
+            Arguments = "new console -f net7.0"
+        };
+
+        // Create the project to pack
+        Process dotnetNew = Process.Start(info);
+        Assert.IsNotNull(dotnetNew);
+        dotnetNew.WaitForExit();
+        Assert.AreEqual(0, dotnetNew.ExitCode);
+
+        info.Arguments = "build --configuration release";
+
+        Process dotnetPublish = Process.Start(info);
+        Assert.IsNotNull(dotnetPublish);
+        dotnetPublish.WaitForExit();
+        Assert.AreEqual(0, dotnetPublish.ExitCode);
+
+        CreateNewImage task = new CreateNewImage();
+        task.BaseRegistry = "https://mcr.microsoft.com";
+        task.BaseImageName = "dotnet/runtime";
+        task.BaseImageTag = "6.0";
+
+        task.OutputRegistry = "http://localhost:5010";
+        task.PublishDirectory = Path.Combine(newProjectDir.FullName, "bin", "release", "net7.0");
+        task.ImageName = "dotnet/testimage";
+        task.WorkingDirectory = "app/";
+        task.Entrypoint = "dotnet build";
+        task.EntrypointArgs = "";
+
+        Assert.IsTrue(task.Execute());
+        newProjectDir.Delete(true);
+    }
+
+    [TestMethod]
+    public void ParseContainerProperties_EndToEnd()
+    {
+        DirectoryInfo newProjectDir = new DirectoryInfo(Path.Combine(Path.GetTempPath(), nameof(ParseContainerProperties_EndToEnd)));
+
+        if (newProjectDir.Exists)
+        {
+            newProjectDir.Delete(recursive: true);
+        }
+
+        newProjectDir.Create();
+
+        ProcessStartInfo info = new ProcessStartInfo
+        {
+            WorkingDirectory = newProjectDir.FullName,
```

Comment:
I made a helper class for this!

```suggestion
        using TransientTestFolder newProjectDir = new(nameof(ParseContainerProperties_EndToEnd));

        ProcessStartInfo info = new ProcessStartInfo
        {
            WorkingDirectory = newProjectDir.Path,
```

(then I forgot about it and didn't use it in some other tests 🙃)

Reply:
Will need to bring that in once it's in main

Reply:
it is, I put it in a long time ago. Doesn't have to block this though

Reply:
ah I didn't see the string parameter and figured it was in some branch that hadn't merged yet.